lib/manifest/PlaylistGenerator.js

```diff
@@ -73,10 +81,16 @@ function savePlaylist() {
         .then(function() {
             that.lastPlaylistUpdate = new Date();
             that.logger.debug("Playlist written to disk");
-            backendClient.updateStreamInfo(that.entryId, that.entryServerNodeId, that.flavorsObjArray)
-                .then(function() {
-                    that.logger.debug("Flavors updated in DB");
-                });
+            if(that.entryServerNodeId) {
```

Comment:
is this nicer?
 var getLiveEntryServerNodeIdPromise=that.entryServerNodeId ? Promise.resolve(that.entryServerNodeId) :  backendClient.getLiveEntryServerNodeId(that.entryId, that.entryServerType);

getLiveEntryServerNodeIdPromise.then(function(entryServerNodeId) {
                         that.entryServerNodeId = entryServerNodeId;
                         updateStreamInfo.call(that);
                     });